client/asset/btc/btc.go

```diff
@@ -1959,6 +2022,24 @@ func (btc *ExchangeWallet) createSig(tx *wire.MsgTx, idx int, pkScript []byte, a
 	return sig, privKey.PubKey().SerializeCompressed(), nil
 }
 
+// createWitnessSig creates and returns a signature for the witness of a segwit
+// input and the pubkey associated with the address.
+func (btc *ExchangeWallet) createWitnessSig(tx *wire.MsgTx, idx int, pkScript []byte,
+	addr btcutil.Address, val uint64, sigHashes *txscript.TxSigHashes) (sig, pubkey []byte, err error) {
+
+	privKey, err := btc.wallet.PrivKeyForAddress(addr.String())
+	if err != nil {
+		return nil, nil, err
+	}
+	sig, err = txscript.RawTxInWitnessSignature(tx, txscript.NewTxSigHashes(tx), idx,
```

Comment:
Did you intend to use the `sigHashes` input parameter instead of calling `txscript.NewTxSigHashes`?